kv() Pebble function throws unexpected exception when trying to access expired key

### Describe the issue

Hey, I encountered this weird / unexpected behavior (not sure if bug, maybe not, but unexpected behaviour, for me at least).

```
Caused by: io.pebbletemplates.pebble.error.PebbleException: The requested value has expired ({{ ['some_table'] + (inputs.dq_excluded_tables | default([])) + (kv('dq_excluded_tables') | default([])) }}
```

This Pebble expression
`{{ ['some_table'] + (inputs.dq_excluded_tables | default([])) + (kv('dq_excluded_tables') | default([])) }}`

works as expected when the KV key doesn't exist (it defaults to an empty array, which is what we want).

But the first time the KV key is accessed after it expires, it throws the above exception. (Subsequent accesses work as the key got deleted when the exception was thrown, and so the default value filter is then used.) This is (I assume) the code responsible for it:
https://github.com/kestra-io/kestra/blob/develop/core/src/main/java/io/kestra/core/storages/kv/InternalKVStore.java#L105

So my issue with this behavior is that from the docs/UI, I would expect the expiration to be a standard behavior = when I set a key to expire after say 7 days, it is correct and expected that it expires and is no longer there. So I don't expect an exception in such case, I expect the `kv()` function to return NULL, same as if the key doesn't exist - which is what I expect, when it expires, it shouldn't exist.

So **proposing to change the behavior to reflect that expectation = when a key expires, it is automatically deleted (possibly soft deleted), no exception is thrown, and NULL is returned because the key no longer exists.** (Or alternatively make the behavior configurable with some parameter)

cc @loicmathieu 

### Example

Example flow that reproducibly demonstrates this issue:

```
id: test-kv-expiration-error
namespace: test.joe
tasks:
  - id: kv_set
    type: io.kestra.plugin.core.kv.Set
    key: test_var
    value: "test_value"
    kvType: STRING
    ttl: PT1S

    # will pause the flow; manually Resume in Kestra UI after 1+ seconds
    # note: Sleep task is better if available in your version of Kestra
  - id: pause
    type: io.kestra.plugin.core.flow.Pause

  - id: "log"
    type: "io.kestra.plugin.core.log.Log"
    level: DEBUG
    message: "{{ kv('test_var') }}"

```

### Environment

- Kestra Version: 0.19.10 (but it looks like the behavior is the same also in the most recent version, possibly with setting errorOnMissing=false to account for this change: https://kestra.io/docs/migration-guide/0.22.0/kv-error-on-missing)

--- a/core/src/main/java/io/kestra/core/runners/pebble/functions/KvFunction.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/functions/KvFunction.java
@@ -38,7 +38,7 @@ public class KvFunction implements Function {
         String key = getKey(args, self, lineNumber);
         String namespace = (String) args.get(NAMESPACE_ARG);
 
-        Boolean errorOnMissing = Optional.ofNullable((Boolean) args.get(ERROR_ON_MISSING_ARG)).orElse(true);
+        boolean errorOnMissing = Optional.ofNullable((Boolean) args.get(ERROR_ON_MISSING_ARG)).orElse(true);
 
         Map<String, String> flow = (Map<String, String>) context.getVariable("flow");
         String flowNamespace = flow.get(NAMESPACE_ARG);
@@ -53,11 +53,16 @@ public class KvFunction implements Function {
                 // we didn't check allowedNamespace here as it's checked in the kvStoreService itself
                 value = kvStoreService.get(flowTenantId, namespace, flowNamespace).getValue(key);
             }
+        } catch (ResourceExpiredException e) {
+            if (errorOnMissing) {
+                throw new PebbleException(e, e.getMessage(), lineNumber, self.getName());
+            }
+            value = Optional.empty();
         } catch (Exception e) {
             throw new PebbleException(e, e.getMessage(), lineNumber, self.getName());
         }
 
-        if (value.isEmpty() && errorOnMissing == Boolean.TRUE) {
+        if (value.isEmpty() && errorOnMissing) {
             throw new PebbleException(null, "The key '" + key + "' does not exist in the namespace '" + namespace + "'.", lineNumber, self.getName());
         }
 

--- a/core/src/test/java/io/kestra/core/runners/pebble/functions/KvFunctionTest.java
+++ b/core/src/test/java/io/kestra/core/runners/pebble/functions/KvFunctionTest.java
@@ -1,25 +1,25 @@
 package io.kestra.core.runners.pebble.functions;
 
-import static io.kestra.core.runners.pebble.functions.FunctionTestUtils.getVariables;
-import static io.kestra.core.tenant.TenantService.MAIN_TENANT;
-import static org.assertj.core.api.Assertions.assertThat;
-
 import io.kestra.core.exceptions.IllegalVariableEvaluationException;
 import io.kestra.core.junit.annotations.KestraTest;
 import io.kestra.core.runners.VariableRenderer;
-import io.kestra.core.storages.StorageContext;
 import io.kestra.core.storages.StorageInterface;
 import io.kestra.core.storages.kv.InternalKVStore;
+import io.kestra.core.storages.kv.KVMetadata;
 import io.kestra.core.storages.kv.KVStore;
 import io.kestra.core.storages.kv.KVValueAndMetadata;
 import io.kestra.core.utils.TestsUtils;
 import jakarta.inject.Inject;
-import java.io.IOException;
-import java.net.URI;
-import java.util.Map;
 import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+
+import static io.kestra.core.runners.pebble.functions.FunctionTestUtils.getVariables;
+import static org.assertj.core.api.Assertions.assertThat;
 
 @KestraTest(startRunner = true)
 public class KvFunctionTest {
@@ -109,6 +109,25 @@ public class KvFunctionTest {
     }
 
     @Test
+    void shouldThrowOrGetEmptyIfExpiredDependingOnErrorOnMissing() throws IOException, IllegalVariableEvaluationException {
+        String tenant = TestsUtils.randomTenant();
+        String namespace = TestsUtils.randomNamespace();
+        Map<String, Object> variables = getVariables(tenant, namespace);
+
+        KVStore kv = new InternalKVStore(tenant, namespace, storageInterface);
+        kv.put("my-expired-key", new KVValueAndMetadata(new KVMetadata(null, Instant.now().minus(1, ChronoUnit.HOURS)), "anyValue"));
+
+        String rendered = variableRenderer.render("{{ kv('my-expired-key', errorOnMissing=false) }}", variables);
+        assertThat(rendered).isEqualTo("");
+
+        kv.put("another-expired-key", new KVValueAndMetadata(new KVMetadata(null, Instant.now().minus(1, ChronoUnit.HOURS)), "anyValue"));
+
+        IllegalVariableEvaluationException exception = Assertions.assertThrows(IllegalVariableEvaluationException.class, () -> variableRenderer.render("{{ kv('another-expired-key') }}", variables));
+
+        assertThat(exception.getMessage()).isEqualTo("io.pebbletemplates.pebble.error.PebbleException: The requested value has expired ({{ kv('another-expired-key') }}:1)");
+    }
+
+    @Test
     void shouldFailGivenNonExistingKeyAndErrorOnMissingTrue() {
         // Given
         String tenant = TestsUtils.randomTenant(this.getClass().getSimpleName());
@@ -129,9 +148,7 @@ public class KvFunctionTest {
         String tenant = TestsUtils.randomTenant(this.getClass().getSimpleName());
         Map<String, Object> variables = getVariables(tenant, "io.kestra.tests");
         // When
-        IllegalVariableEvaluationException exception = Assertions.assertThrows(IllegalVariableEvaluationException.class, () -> {
-            variableRenderer.render("{{ kv('my-key') }}", variables);
-        });
+        IllegalVariableEvaluationException exception = Assertions.assertThrows(IllegalVariableEvaluationException.class, () -> variableRenderer.render("{{ kv('my-key') }}", variables));
 
         // Then
         assertThat(exception.getMessage()).isEqualTo("io.pebbletemplates.pebble.error.PebbleException: The key 'my-key' does not exist in the namespace 'io.kestra.tests'. ({{ kv('my-key') }}:1)");

